File: tests/fixtures/kotlin-library/src/main/kotlin/library/models/Book.kt
package library.models

/** Maximum number of search results to return. */
const val MAX_RESULTS: Int = 100

/** A book in the library catalog. */
data class Book(
    val title: String,
    val isbn: String,
    val genre: Genre,
    val copiesAvailable: Int = 1
) {
    /** Check if the book is available for borrowing. */
    fun isAvailable(): Boolean = copiesAvailable > 0

    /** Extension: companion object with factory methods. */
    companion object {
        fun create(title: String, isbn: String): Book =
            Book(title, isbn, Genre.FICTION)

        fun fromJson(json: String): Book =
            Book("Parsed", "000-0", Genre.FICTION)
    }
}

File: tests/fixtures/rust-library/src/models/book.rs
/// A book in the library catalog.
pub struct Book {
    title: String,
    isbn: String,
    genre: super::genre::Genre,
    copies_available: u32,
}

/// Maximum number of search results to return.
pub const MAX_RESULTS: usize = 100;

impl Book {
    /// Create a new book.
    pub fn new(title: String, isbn: String, genre: super::genre::Genre) -> Self {
        Self {
            title,
            isbn,
            genre,
            copies_available: 1,
        }
    }

    /// Get the book title.
    pub fn title(&self) -> &str {
        &self.title
    }

    /// Get the ISBN.
    pub fn isbn(&self) -> &str {
        &self.isbn
    }

    /// Check if the book is available for borrowing.
    pub fn is_available(&self) -> bool {
        self.copies_available > 0
    }

    /// Get the genre.
    pub fn genre(&self) -> &super::genre::Genre {
        &self.genre
    }
}
